ludwig/config_validation/validation.py

```diff
@@ -28,11 +28,19 @@ def get_schema(model_type: str = MODEL_ECD):
 
     cls = model_type_schema_registry[model_type]
     props = unload_jsonschema_from_marshmallow_class(cls)["properties"]
+
+    # TODO: Replace with more robust required logic later.
+    required = []
+    if model_type == MODEL_ECD or model_type == MODEL_GBM:
+        required = ["input_features", "output_features"]
+
     return {
         "type": "object",
         "properties": props,
         "title": "model_options",
         "description": "Settings for Ludwig configuration",
+        "required": required,
+        "additionalProperties": True,  # TODO: Set to false after 0.8 releases.
```

Comment:
@tgaddair Actually, do we want to make this change for 0.8 or do it after?

Reply:
Making input and output features required fields? I'd say make this change now.

Reply:
@justinxzhao Ah sorry, I should have been clearer. Do we want to set `additionalProperties` to false now or later?

Reply:
Ah I see.

Curious to know what other people think. I'd be in favor of making the change now since 1) there are robustness benefits, 2) we've wanted to make this change for a while, 3) there's an easy way to add a new parameter with schema_utils.Dict().

Is the error message for legacy users who might have inadvertently been including extra parameters in their configs clear?

Reply:
@justinxzhao This is the deprecation warning that shows up right now:
https://github.com/ludwig-ai/ludwig/blob/dc0d685ac70729c667acc83f85c897cab7b38917/ludwig/schema/utils.py#L186

It actually mentions that 0.8 will be strict, so I would like to do this now too.

In any case, I'll raise a separate PR for it and we can have the discussion there if need be.